### DISCUSSION TITLE
Compilation Error  when trying to build Moltres on my Ubuntu system

### DISCUSSION THREAD

**Discussion by hityyds** [arfc/moltres#280] [Q&A Getting started] [answered]
Hello everyone,

I'm encountering a compilation error when trying to build Moltres on my Ubuntu system. The error seems to be related to the `PetscErrorCode` type, where the code uses an `int` type instead of the expected `PetscErrorCode` type. Here is a snippet of the error message I’m receiving:
```
~/ProgramFiles/Moltres_install/moltres/moose/modules/navier_stokes/src/problems/NavierStokesProblem.C:98:25: error: invalid conversion from 'int' to 'PetscErrorCode' [-fpermissive]
   98 |   PetscErrorCode ierr = 0;
      |                         ^
      |                         |
      |                         int
```

Additionally, when I started compiling, the system gave the following prompt:
```
~/ProgramFiles/Moltres_install/moltres$ make -j8
Not compiling MOOSE with NEML2 because NEML2_DIR is not a valid NEML2 checkout and/or libtorch is not enabled.
Rebuilding symlinks in ~/ProgramFiles/Moltres_install/moltres/moose/framework/build/header_symlinks
```

I have already installed the MOOSE Conda environment:
```
~/ProgramFiles/Moltres_install/moltres$ conda list | grep moose
moose-dev                 2024.08.26                mpich    https://conda.software.inl.gov/public
moose-libmesh             2024.08.17              mpich_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.3.0                   mpich_2    https://conda.software.inl.gov/public
moose-mpi                 2024.08.12                mpich    https://conda.software.inl.gov/public
moose-peacock             2024.08.12              mpich_0    https://conda.software.inl.gov/public
moose-petsc               3.21.4                  mpich_1    https://conda.software.inl.gov/public
moose-tools               2024.07.19      py311h6612bce_0    https://conda.software.inl.gov/public
moose-wasp                2024.05.08              build_0    https://conda.software.inl.gov/public
```

Does anyone know how to solve this? I noticed that the master branch hasn’t been updated in six months. Could this be causing incompatibility issues with the latest MOOSE version? Any advice or insights would be greatly appreciated. Thank you in advance!

**Comment by smpark7** [comment, marked as answer]
We're currently facing some compatibility issues with the latest MOOSE environment. I recommend using moose-dev=2024.03.05 with the current master branch or moose-dev=2024.07.19 with the MOOSE version in PR #279.

**Reply by smpark7** [reply]
As for NEML2, that's an optional dependency. We don't currently use it
